zarr/tests/test_core.py

```diff
@@ -2215,16 +2127,7 @@ def expected(self):
 
 class TestArrayWithBloscCompressor(TestArray):
 
-    def create_array(self, read_only=False, **kwargs):
-        store = KVStore(dict())
-        compressor = Blosc(cname='zstd', clevel=1, shuffle=1)
-        kwargs.setdefault('compressor', compressor)
-        cache_metadata = kwargs.pop('cache_metadata', True)
-        cache_attrs = kwargs.pop('cache_attrs', True)
-        write_empty_chunks = kwargs.pop('write_empty_chunks', True)
-        init_array(store, **kwargs)
-        return Array(store, read_only=read_only, cache_metadata=cache_metadata,
-                     cache_attrs=cache_attrs, write_empty_chunks=write_empty_chunks)
+    compressor = Blosc(cname="zstd", clevel=1, shuffle=1)
```

Comment:
Feels like we should be using [parametrized fixtures](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) here rather than making a class for each compressor.